File: competence-database/src/main/scala/uzuzjmd/competence/owl/dao/Comment.scala
package uzuzjmd.competence.owl.dao

import uzuzjmd.competence.owl.ontology.CompOntClass
import uzuzjmd.competence.owl.access.CompOntologyManager
import uzuzjmd.competence.owl.ontology.CompObjectProperties
import uzuzjmd.competence.owl.access.CompOntologyAccess

case class Comment(comp: CompOntologyManager, val text: String, val creator: User = null, val created: java.lang.Long = null, val textReadable : String = null) extends CompetenceOntologyDao(comp, CompOntClass.Comment, text) {

  def DATECRATED = "datecreated"
  def TEXT = "text"

  @Override
  protected def deleteMore() {
    //TODO

  }

  @Override
  protected def persistMore() {
    creator.persist
    createEdgeWith(creator, CompObjectProperties.UserOfComment)    
    addDataField(TEXT, identifierBeforeParsing)
    if (created == null) {
    	addDataField(DATECRATED, System.currentTimeMillis(): java.lang.Long);
    } else {
      addDataField(DATECRATED, created);
    }
  }

  def getFullDao(): Comment = {
    val creatorNew = getAssociatedStandardDaosAsDomain(CompObjectProperties.UserOfComment, classOf[User]).head
    val textNew = getDataField(TEXT)
    val dateCreatedNew = getDataFieldLong(DATECRATED)
    return new Comment(comp, text, creatorNew, dateCreatedNew, textNew)
  }
}

File: competence-database/src/main/scala/uzuzjmd/competence/monopersistence/daos/Comment.java
package uzuzjmd.competence.monopersistence.daos;

import uzuzjmd.competence.exceptions.NoUserGivenException;
import uzuzjmd.competence.monopersistence.Cascadable;
import uzuzjmd.competence.monopersistence.DaoAbstractImpl;
import uzuzjmd.competence.persistence.ontology.CompObjectProperties;

/**
 * Created by dehne on 11.01.2016.
 */
public class Comment extends AbstractComment implements Cascadable {

    private Long dateCreated;
    private String text;

    public Comment(String id) {
        super(id);
    }

    public Comment(String text, User creator, Long dateCreated) {
        super(dateCreated + text);
        this.creator = creator;
    }

    public Long getDateCreated() {
        return dateCreated;
    }

    public void setDateCreated(Long dateCreated) {
        this.dateCreated = dateCreated;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    @Override
    public void persistMore() throws Exception {
        this.persist();
        if (creator == null) {
            throw new NoUserGivenException();
        }
        createEdgeWith(creator, CompObjectProperties.UserOfComment);
    }




}
